moveit_py/moveit/policies/policy.py

```diff
@@ -0,0 +1,147 @@
+# Software License Agreement (BSD License)
+#
+# Copyright (c) 2023, Peter David Fagan
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions
+# are met:
+#
+#  * Redistributions of source code must retain the above copyright
+#    notice, this list of conditions and the following disclaimer.
+#  * Redistributions in binary form must reproduce the above
+#    copyright notice, this list of conditions and the following
+#    disclaimer in the documentation and/or other materials provided
+#    with the distribution.
+#  * Neither the name of Willow Garage, Inc. nor the names of its
+#    contributors may be used to endorse or promote products derived
+#    from this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+# "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+# LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+# FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+# COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+# INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+# BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+# LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+# LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+# ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+#
+# Author: Peter David Fagan
+
+""" Definition of an abstract base class for policy deployment. """
```

Comment:
You may want to mention that this is only for Pose commands. Currently, a lot of MoveIt users are accustomed to trajectories.

```suggestion
""" Definition of an abstract base class for policy deployment. The commands are pose-based. """
```

Reply:
Oh actually I need to add `jointjog` [here](https://github.com/peterdavidfagan/moveit2/blob/e9986e9523dbe064eb24095e11c36a36359811ea/moveit_py/moveit/policies/policy.py#L99).

Reply:
ok. Not trajectories, though

Reply:
Ah my apologies I read you question too quickly, you want to highlight the difference between this and `RobotTrajectory`. I am happy to add this. 

Part of me also believes we may want to expose motion planning to the policies submodule in future. Do you agree?

Reply:
yep

Reply:
[89160f4](https://github.com/ros-planning/moveit2/pull/2494/commits/89160f4e5686afa3d6a1cbc602957fb4e0301ee6)